test/unit/ui/map.test.js

```diff
@@ -785,8 +785,13 @@ test('Map', (t) => {
     t.test('#remove', (t) => {
         const map = createMap(t);
         t.equal(map.getContainer().childNodes.length, 3);
+        // LogoControl is added to the map in the constructor
```

Comment:
How about something like:

```
t.test('#remove calls onRemove on added controls', (t) => {
    const map = createMap(t);
    const control = { onRemove: t.spy() };
    map.addControl(control);
    map.remove();
    t.ok(control.onRemove.calledOnce);
});
```

This avoids relying on map implementation details, and tests one desired behavior in isolation from others so it's easier to understand and debug if it fails.